solutions/cses-1081.mdx

```diff
@@ -44,6 +46,55 @@ int main() {
 	cout << ans << endl;
 }
 ```
+</CPPSection>
+
+<JavaSection>
+```java
+import java.util.Scanner;
+
+public class Main {
+	public static int gcd(int a, int b) {
+		return b == 0 ? a : gcd(b, a % b);
+	}
+
+	public static void main(String[] args) {
+		Scanner scanner = new Scanner(System.in);
```

Comment:
use bufferedreader

Reply:
oops forgot that existed 💀
bufferedreader does a lot better, but three cases still tle -- couldnt seem to figure out a way to get it to ac